string.c

```diff
@@ -3484,15 +3484,20 @@ str_casecmp_p(VALUE str1, VALUE str2)
     VALUE folded_str1, folded_str2;
     VALUE fold_opt = sym_fold;
 
-    enc = rb_enc_compatible(str1, str2);
-    if (!enc) {
-	return Qnil;
+    if (rb_enc_str_asciionly_p(str1)) {
+        return FIXNUM_ZERO_P(str_casecmp(str1, str2)) ? Qtrue : Qfalse;
```

Comment:
Looks like this would change the returning value from `Qnil` to `Qfalse` when `rb_enc_compatible(str1, str2)` is not true. Should we have a test for that case?

Reply:
Thanks for your review. I added a test case and fixed the point.